wit/couchbase/types.wit

```diff
@@ -0,0 +1,145 @@
+package wasmcloud:couchbase@0.1.0-draft;
+
+/// General types that can be used throughout the couchbase interface
+interface types {
+  /// ID of a Document
+  ///
+  /// Note that Document IDs are expected to be:
+  ///   - Assigned at the application level
+  ///   - Be valid UTF-8 at the application level
+  ///   - Be no longer than 250 bytes (not characters)
+  type document-id = string;
+
+  /// Path to a subdocument inside an existing document
+  type subdocument-path = string;
+
+  /// The span of a request, normally used when performing/enabling tracing
+  type request-span = string;
+
+  /// A string that is properly formatted JSON
+  type json-string = string;
+
+  /// Document field name (ex. to include in a search)
+  type document-field-name = string;
+
+  /// Collection name to include in a search
+  type collection-name = string;
+
+  /// Name of a search index
+  type search-index-name = string;
+
+  /// Direction in which to perform sorting
+  enum sort-direction {
+    asc,
+    desc,
+  }
+
+  /// Errors that can occur during operations on documents
+  variant document-error {
+    /// Document could not be found
```

Comment:
Regarding CAS here, we should add an error type that we can return when optimistic locking fails.

Reply:
Hey which one of [these errors](https://github.com/couchbase/gocb/blob/v2.8.1/error.go) would that be? I didn't add absolutely all of them but I'm assuming the case you're talking about maps to one of them

Reply:
[This one](https://github.com/couchbase/gocb/blob/v2.8.1/error.go#L75) looks like it fits, so going with it --